include/deal.II/base/tensor_accessors.h

```diff
@@ -483,7 +484,7 @@ namespace TensorAccessors
     // subsequently stored indices:
 
     template <int rank, typename S>
-    class StoreIndex : private S
+    class StoreIndex : public S
```

Comment:
This, by the way, strikes me as odd. I don't know where you use this class, but does it model an "is-a" relationship? If not, maybe using a member variable of type `S` would be more appropriate.